WIP: Add documentation per fragment

mpourismaiel opened the pull request:
**What this PR does / why we need it**:
Write documentation for each fragment.

<!--
**Which issue this PR fixes** *(optional - uncomment and add issue)*:
fixes #0
-->

**Special notes for your reviewer**:
**WORK IN PROGRESS**

Todo:

- [x] Document global variables
- [x] Document asset object
- [ ] Document fragments
  - [x] 404
  - [x] content
  - [x] faq	
  - [x] hero
  - [x] member
  - [x] react-portal
  - [x] buttons
  - [x] copyright
  - [x] footer
  - [ ] item
  - [x] nav
  - [x] search
  - [ ] config
  - [x] editor
  - [x] graph
  - [ ] items
  - [x] portfolio
  - [x] table
  - [x] contact
  - [x] embed
  - [x] header
  - [x] list
  - [x] pricing
  - [x] toc
  - [ ] stripe

**Release note**:
```release-note
- Update documentation
- Add documentation for each fragment
```

stp-ip commented:
Should be able to finally finish this this week. Sorry or the delay.

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+*type: boolean*  
+*default: false*
+
+If set to `true`, a text input would be added to the navbar and changes the layout of menus. The input is used for search.
```
```suggestion
If set to `true`, search is enabled within the navbar.  

**Note:** The additional input field used for search will alter the menu layout.
```

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+If object is set, breadcrumbs will be shown under the navbar.
+
+##### breadcrumb.display
+*type: boolean*
```
```suggestion
*type: boolean*  
```

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+
+##### breadcrumb.display
+*type: boolean*
+*default: in case object is defined the value is true, if not, it's false*
```
What are you trying to say?

mpourismaiel replied:
Now that I think about it, it's unnecessary. I was trying to say that breadcrumb wouldn't display unless the object is defined.

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+*default: in case object is defined the value is true, if not, it's false*
+
+##### breadcrumb.level
+*type: number*
```
```suggestion
*type: number*  
```

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+*type: number*
+*default: 1*
+
+Specifies from which level of nested directories the breadcrumb should be shown. For example for the value of 1:
```
```suggestion
Define the section level the breadcrumb will start being shown.
 
The default value `1` would lead to the following being defined: 
```

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
````diff
@@ -0,0 +1,80 @@
+```
+
+##### breadcrumb.background
+*type: primary, secondary, light, dark, warning, success (other Bootstrap background colors are acceptable as well)*  
````
```suggestion
*type:  string*  
*recommended: primary, secondary, white, light, dark*  
*accepted values: primary, secondary, white, light, dark, warning, success, danger, info, transparent*
```

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+#### asset
+*type: asset object*
+
+Asset will be shown as logo.
```
```suggestion
Asset will be shown as a clickable logo directing to the main page.
```

stp-ip replied:
Is that actually always the case? Do we overwrite this, when asset.url is provided?

mpourismaiel replied:
It's always overwriting the url.

stp-ip commented on exampleSite/content/fragments/nav/docs.md:
```diff
@@ -0,0 +1,80 @@
+#### repo_button
+*type: object*
+
+If set, a repo button would be added to the navbar much like the Github button on top of this page.
```
```suggestion
Enable a button on the top right navbar. Usually used to link to your repository such as Github or Gitlab.  
The icon can be customized via `repo_button.icon`.
```

stp-ip commented on exampleSite/content/fragments/react-portal/docs.md:
```diff
@@ -0,0 +1,26 @@
+
+*Replace capitalized phrases in the code above with what you need.*
+
+The script above needs to be registered in the page where you have the React portal fragment. You can make use of Config fragment to either directly add the code in a `script` tag or load your JS file (recommended). You can also register your JS file as a custom JS file which would load the file in every page which is not recommended unless you're developing a single page application.
```
I think this needs clarification. Hard to follow the thought here. But maybe that's just me.

mpourismaiel commented on exampleSite/content/fragments/buttons/docs.md:
```diff
@@ -0,0 +1,36 @@
+*type: string*
+*required*
+
+Title of the call to action button.
```
In mostly React UI frameworks the prop for it is title. Do you have anything in mind?

stp-ip replied:
Just seemed strange, that the field is called .text, but we describe it as the title of the button.

mpourismaiel commented on exampleSite/content/fragments/buttons/docs.md:
```diff
@@ -0,0 +1,36 @@
+
+##### buttons.url
+*type: string*
+*default: "#"*
```
I don't think I understand. Could you elaborate?

mpourismaiel commented on exampleSite/content/docs/global-variables/content.md:
```diff
@@ -19,15 +31,47 @@ For the background colors of `white`, `light`, `secondary` and `primary` we use
+#### asset
+*type: asset object*
+
+This variable is not global and is not always named `asset`. Any fragment that uses this variable would show either an image or an icon.
```
Changed to:

```
This variable is not a global variable but a variable type that is used in a lot of fragment.

Any fragment that uses this variable type would show either an image or an icon.

The type is introduced to make configuring images and icons same between different fragments.
```

Is that better?

mpourismaiel commented:
I updated using the suggestions. Will follow up on others presently.

mpourismaiel commented on exampleSite/content/fragments/404/docs.md:
```diff
@@ -0,0 +1,40 @@
+#### asset
+*type: asset object*
+
+The asset will be displayed on top of the fragment, before title and subtitle.
```
Icons is not supported here.

stp-ip commented on exampleSite/content/docs/global-variables/content.md:
```diff
@@ -54,24 +55,29 @@ Changes the padding of fragment's container
 *type: asset object*
 
-This variable is not global and is not always named `asset`. Any fragment that uses this variable would show either an image or an icon.
+This variable is not a global variable but a variable type that is used in a lot of fragment.
```
```suggestions
This variable is not a global variable but a variable type that is used in a lot of fragments.
```

stp-ip commented:
Resolved comments that got fixed. Makes me happy every time I'm taking a look at the revamped docs page \o/

mpourismaiel commented on exampleSite/content/fragments/contact/docs.md:
```diff
@@ -0,0 +1,124 @@
+
+You can use this array to add new hidden fields to the form.
+
+##### fields.hidden.name
```
I think the value for `site` is wrong. `$.root.Page.Permalink` should output the permalink for the current page. Should we change that or is that intended? I'll document the current behavior.

mpourismaiel commented:
I think you missed some of the resolved comments. Some of them are marked `outdated` without extra comments. Are there any problems?

Also updated some small parts, added documentation for toc fragment and merged master into the branch.

stp-ip commented:
The outdated ones seemed to be resolved in my view. If I missed something let me know.

I think we can close this one after we add a placeholder similar to stripe for item, items and config. I can do the docs for these next week in a follow on PR.
Sounds good?

mpourismaiel commented:
Sounds great. Added the placeholders.
